### PR TITLE
Add utility function that displays strings

### PR DESCRIPTION
**Pull request by fpsunflower**
Based on the previous review that added `getchar(string, index)` and `hash(string)`, add a new utility function in `oslutil.h` that lets a shader draw a string.

This is convenient for displaying text on objects in raster space for example.

### CODE REVIEW

**Comment by lgritz** [comment]
This is insane and I love it.
LGTM

File: testsuite/draw_string/run.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env python
+
+command += testshade ("-g 512 512 -od uint8 -o Cout out.tif test")
+outputs = [ "out.txt", "out.tif" ]
+# expect a few LSB failures
+failthresh = 0.004
+failpercent = 0.05
```
Comment (lgritz) [inline testsuite/draw_string/run.py]: A few tests need slightly higher fail thresholds, but if this passes without these two lines, please take them out.

**Comment by fpsunflower** [comment]
Yup it seems to pass fine without the thresholds. Let's see how it does on the remote hosts ...

**Comment by fpsunflower** [comment]
Passed the CI tests as well. Merged.

**Pull request merged**